duktape/src/main/jni/JavaMethod.cpp

```diff
@@ -76,6 +86,46 @@ jobject getPrimitiveType(JNIEnv* env, const char* boxedClassName) {
 }
 
 /**
+ * Pops the value from the top of the stack using {@code popPrimitive}, then
+ * boxes it in an object of class {@code type} by calling {@code constructor}.
+ */
+jvalue boxPrimitive(duk_context* ctx, JNIEnv* env, jclass type, jmethodID constructor,
+    std::function<jvalue(duk_context*, JNIEnv*)> popPrimitive) {
+  // Check if the caller passed in a null value.
+  if (duk_get_type(ctx, -1) == DUK_TYPE_NULL) {
+    duk_pop(ctx);
+    jvalue value;
+    value.l = nullptr;
+    return value;
+  }
+
+  jvalue value = popPrimitive(ctx, env);
+  value.l = env->NewObjectA(type, constructor, &value);
```

Comment:
Can we call the static `valueOf` method that all boxed types have here? That way we leverage their individual caches for small/some values instead of actually allocating each time.